### PR TITLE
Feature/ab#62 and ab#63 add groups

### PR DESCRIPTION
**Pull request by witchpou**
It should be possible to separate apps and appTemplates by group

## Description

see ticket ab#62 and ab#63

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: webclient/app/src/app/commons/appTemplateDialog/ChipSelect.jsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+  const handleChange = (event) => {
+    const {
+      target: { value },
+    } = event;
+    setItemValue(
+      // On autofill we get a stringified value.
+      typeof value === 'string' ? value.split(',') : value,
+    );
+    handleExternalChange(typeof value === 'string' ? value.split(',') : value,);
```
Comment (DerCed) [inline webclient/app/src/app/commons/appTemplateDialog/ChipSelect.jsx:22]: is ``value,`` correct?

File: webclient/app/src/app/commons/appTemplateDialog/ChipSelect.jsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        <InputLabel id="demo-multiple-chip-label">{label}</InputLabel>
+        <Select
+          labelId="demo-multiple-chip-label"
+          id="demo-multiple-chip"
+          multiple
+          value={itemValue}
+          onChange={handleChange}
+          input={<OutlinedInput id="select-multiple-chip" label={label} />}
+          renderValue={() => (
+            <Box sx={{ display: 'flex', flexWrap: 'wrap', gap: 0.5 }}>
```
Comment (DerCed) [inline webclient/app/src/app/commons/appTemplateDialog/ChipSelect.jsx:40]: Inline Styles

File: webclient/app/src/app/pages/appTemplate/AppTemplateOverview.jsx [hunk 3]
```diff
@@ -35,6 +38,13 @@ function AppTemplateOverview() {
         });
     };
 
+    useEffect(() => {
+        const userRest = new UserRest();
```
Comment (DerCed) [inline webclient/app/src/app/pages/appTemplate/AppTemplateOverview.jsx:42]: I would prefer to create one instance of UserRest() on the top with the use of useMemo

**Review by DerCed — CHANGES_REQUESTED** [review]
Im missing context here. By this PR information (+ tickets) I cannot clearly say, what this ticket should do

File: webclient/app/src/app/commons/appTemplateDialog/ChipSelect.jsx (hunk 1 above)
Comment (witchpou) [reply webclient/app/src/app/commons/appTemplateDialog/ChipSelect.jsx:22]: was changed

File: webclient/app/src/app/commons/appTemplateDialog/ChipSelect.jsx (hunk 2 above)
Comment (witchpou) [reply webclient/app/src/app/commons/appTemplateDialog/ChipSelect.jsx:40]: changed

File: webclient/app/src/app/pages/appTemplate/AppTemplateOverview.jsx (hunk 3 above)
Comment (witchpou) [reply webclient/app/src/app/pages/appTemplate/AppTemplateOverview.jsx]: done also for GeneralSection.jsx

File: webclient/app/src/app/commons/multipleSelectChip/MultipleSelectChip.jsx [hunk 4]
```diff
@@ -0,0 +1,58 @@
+import React, { useEffect, useState } from "react";
```
Comment (DerCed) [inline webclient/app/src/app/commons/multipleSelectChip/MultipleSelectChip.jsx:1]: MultipleSelectChip might be a misleading name. Maybe MultipleChipSelect would be a better one

File: webclient/app/src/app/commons/multipleSelectChip/MultipleSelectChip.jsx [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+  useEffect(() => {
+    setItemValue(selected);
+  }, [selected]);
+
+  return (
+      <FormControl fullWidth>
+        <InputLabel id="demo-multiple-chip-label">{label}</InputLabel>
+        <Select
+          labelId="demo-multiple-chip-label"
+          id="demo-multiple-chip"
```
Comment (DerCed) [inline webclient/app/src/app/commons/multipleSelectChip/MultipleSelectChip.jsx:35]: Generic id and labelId

**Review by DerCed — APPROVED** [review]
Please note the points we discussed in our last week

File: webclient/app/src/app/commons/multipleSelectChip/MultipleSelectChip.jsx (hunk 4 above)
Comment (witchpou) [reply webclient/app/src/app/commons/multipleSelectChip/MultipleSelectChip.jsx:1]: Das heißt im Beispiel auch genau so

**Pull request merged**